### PR TITLE
Enable T.38 passthrough in Asterisk

### LINKED ISSUE #957: Enable T.38 passthrough in Asterisk
Currently only virtual faxes initiated (or ended) in Asterisk are supported.

This issue aims to enable T.38 passthrough for vPBX, retail and residential clients.

Retail accounts will need to be marked as "fax" to route traffic through Asterisk.

### PR DESCRIPTION
**Pull request by cruzccl**
This commit fixes #957

### CODE REVIEW

File: asterisk/agi/src/Dialplan/Retails.php [hunk 1]
```diff
@@ -81,17 +81,30 @@ public function process()
         $caller = new RetailAgent($this->agi, $retailAccount);
         $this->channelInfo->setChannelCaller($caller);
 
-        // Only forwarded calls are allowed from retail accounts
-        if ($this->agi->getRedirecting('count') == 0) {
+        // Only forwarded calls or T.38 calls are allowed from retail accounts
+        $isCallForward = $this->agi->getRedirecting('count') != 0;
+        $isFax = $retailAccount->getT38Passthrough() == 'yes';
```
Comment (Kaian) [inline asterisk/agi/src/Dialplan/Retails.php]: Use RetailAccountInterface constants and strict comparison here

File: library/Ivoz/Ast/Domain/Service/PsEndpoint/UpdateByFriend.php [hunk 2]
```diff
@@ -76,8 +76,16 @@ public function execute(FriendInterface $entity)
             ->setDirectmediaMethod($entity->getDirectmediaMethod())
             ->setTrustIdInbound('yes')
             ->setOutboundProxy('sip:users.ivozprovider.local^3Blr')
+            ->setT38Udptl($entity->getT38Passthrough())
             ->setDirectMediaMethod('invite');
 
+        // Disable direct media for T.38 capable devices
+        if ($entity->getT38Passthrough() == 'yes') {
```
Comment (Kaian) [inline library/Ivoz/Ast/Domain/Service/PsEndpoint/UpdateByFriend.php]: Use FriendInterface constants and strict comparison here

File: library/Ivoz/Ast/Domain/Service/PsEndpoint/UpdateByResidentialDevice.php [hunk 3]
```diff
@@ -79,8 +79,16 @@ public function execute(ResidentialDeviceInterface $entity)
             ->setDirectmediaMethod($entity->getDirectmediaMethod())
             ->setTrustIdInbound('yes')
             ->setOutboundProxy('sip:users.ivozprovider.local^3Blr')
+            ->setT38Udptl($entity->getT38Passthrough())
             ->setDirectMediaMethod('invite');
 
+        // Disable direct media for T.38 capable devices
+        if ($entity->getT38Passthrough() == 'yes') {
```
Comment (Kaian) [inline library/Ivoz/Ast/Domain/Service/PsEndpoint/UpdateByResidentialDevice.php]: Use ResidentialDeviceInterface constants and strict comparison here

File: scheme/app/DoctrineMigrations/Version20190423104815.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'mysql', 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE Friends ADD t38Passthrough VARCHAR(255) DEFAULT \'no\' NOT NULL COMMENT \'[enum:yes|no]\'');
```
Comment (Kaian) [inline scheme/app/DoctrineMigrations/Version20190423104815.php]: Setting length to these field string enums definition in ORM files will avoid VARCHAR(255) in database fields

**Comment by cruzccl** [comment]
**This feature breaks IvozProvider design, forcing Application Servers to be exposed** (read documentation).

We don't like this feature, we don't like faxes :(

**Pull request merged**